src/Servers/IIS/IIS/test/Common.FunctionalTests/Infrastructure/EventLogHelpers.cs

```diff
@@ -65,12 +59,27 @@ public static string OnlyOneAppPerAppPool()
         }
     }
 
-    private static EventLogEntry[] AssertEntry(string regexString, IEnumerable<EventLogEntry> entries, bool allowMultiple = false)
+    private static async Task<EventLogEntry[]> AssertEntry(string regexString, IISDeploymentResult deploymentResult, bool allowMultiple = false)
     {
-        var expectedRegex = new Regex(regexString, RegexOptions.Singleline);
-        var matchedEntries = entries.Where(entry => expectedRegex.IsMatch(entry.Message)).ToArray();
-        Assert.True(matchedEntries.Length > 0, $"No entries matched by '{regexString}'");
-        Assert.True(allowMultiple || matchedEntries.Length < 2, $"Multiple entries matched by '{regexString}': {FormatEntries(matchedEntries)}");
+        // EventLogs don't seem to be instant, add retries to attempt to reduce test failures when looking for logs.
+        var retryCount = 10;
+        EventLogEntry[] matchedEntries = [];
+        while (retryCount > 0)
```

Comment:
I'm curious why this isn't a for loop.  Maybe we don't like backwards for loops?